Libraries/Azure.psm1

```diff
@@ -213,6 +213,59 @@ Function Validate-SubscriptionUsage($RGXMLData, $Location, $OverrideVMSize, $Sto
     }
 }
 
+Function Change-StorageAccountType($TestCaseData, [string]$Location, $GlobalConfig, $OsVHD) {
+    $storageAccount = $GlobalConfig.Global.Azure.Subscription.ARMStorageAccount
+    $changedSC = ""
+    $copyVHD = $false
+    if ($TestCaseData.AdditionalHWConfig.SCType -and $TestCaseData.AdditionalHWConfig.SCType.Contains("Premium")) {
+        # if SC is ExistingStorage_Standard, switch to ExistingStorage_Premium
+        if ($storageAccount -imatch "ExistingStorage_Standard") {
+            $changedSC = Get-StorageAccountFromRegion -Region $Location -StorageAccount "ExistingStorage_Premium"
+            # if it is OsVHD format, need copy VHD from standard to premium storage account
+            if ($OsVHD) {
+                $copyVHD = $true
+            }
+        # if SC is NewStorage_Standard, switch to NewStorage_Premium
+        } elseif ($storageAccount -imatch "NewStorage_Standard") {
+            $changedSC = "NewStorage_Premium_LRS"
+        } elseif (($storageAccount -imatch "NewStorage_Premium") -or ($storageAccount -imatch "ExistingStorage_Premium")) {
+            $changedSC = ""
+        } else {
+            $current_sc = Get-StorageAccountFromRegion -Region $Location -StorageAccount "ExistingStorage_Standard"
+            if ($current_sc -eq $storageAccount) {
+                $changedSC = Get-StorageAccountFromRegion -Region $Location -StorageAccount "ExistingStorage_Premium"
+                if ($OsVHD) {
+                    $copyVHD = $true
+                }
+            } else {
+                $storageAccountType = (Get-AzureRmStorageAccount | Where-Object {$_.StorageAccountName -eq $storageAccount}).Sku.Tier.ToString()
+                if ($storageAccountType -inotmatch "premium") {
+                    Write-LogErr "Provided storage account is not premium type, this case $($TestCaseData.testName) need run under premium type of storage account."
+                    Throw "Case $($TestCaseData.testName) need run under premium type of storage account."
+                }
+            }
+        }
+    }
+
+    if ($copyVHD) {
```

Comment:
About copy VHD, we need to make sure that copy operation does not conflict with other copy operations. This could happen, if you start TWO fio tests with same VHD simultaneously.

Reply:
Currently LISAv2 has the same behavior when provided OSVhd the source storage account is not our destination storage account, let's cover this in another PR.

Reply:
@LiliDeng Agreed.